### PR TITLE
Deploy Application keyvault

### PR DESCRIPTION
**Pull request by kbabigumira**
Added the IaC Bicep module and pipeline to deploy Application KeyVault that will be used by the ADP Platform.

### CODE REVIEW

File: infra/keyvault/application-keyvault.bicep [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    name: keyVault.name
+    tags: combinedTags
+    vaultSku: keyVault.skuName
+    enableRbacAuthorization: true    
+    enableSoftDelete: bool(keyVault.enableSoftDelete)
+    enablePurgeProtection: bool(keyVault.enablePurgeProtection)
+    softDeleteRetentionInDays: int(keyVault.softDeleteRetentionInDays)
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Allow'
```
Comment (DaRoszko) [inline infra/keyvault/application-keyvault.bicep:42]: Should be Deny here I think?

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: infra/default-tags.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "ServiceName": "ADP",
+    "ServiceCode" :"CDO"
```
Comment (DaRoszko) [inline infra/default-tags.json:3]: this file exists already doesn't it in the main branch?

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: infra/keyvault/application-keyvault.bicep [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+}
+
+var defaultTags = json(loadTextContent('../default-tags.json'))
+var combinedTags = union(defaultTags, tags, customTags)
+
+module vaults 'br/SharedDefraRegistry:key-vault.vaults:0.5.6' = {
+  name: '${uniqueString(deployment().name, location)}-keyvault'
```
Comment (aadev1) [inline infra/keyvault/application-keyvault.bicep:31]: @kbabigumira @osagiestar - Is it possible to change the deployment name to the convention we've used for AKS and ServiceBus for consistency?

e.g.
@description('Required. Date in the format yyyyMMdd-HHmmss.')
param deploymentDate string = utcNow('yyyyMMdd-HHmmss')

name: 'app-keyvault-${deploymentDate}'

Comment (kbabigumira) [reply infra/keyvault/application-keyvault.bicep]: updated to match the convention used for the other resources

File: infra/keyvault/application-keyvault.bicep (hunk 1 above)
Comment (kbabigumira) [reply infra/keyvault/application-keyvault.bicep]: Changed to Deny

File: infra/default-tags.json (hunk 2 above)
Comment (kbabigumira) [reply infra/default-tags.json:3]: It previously was not there until Asad completed his PR.

Comment (kbabigumira) [reply infra/default-tags.json]: Rebased the branch

**Review by DaRoszko — APPROVED** [review]

File: infra/keyvault/application-keyvault.bicep [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    enableSoftDelete: bool(keyVault.enableSoftDelete)
+    enablePurgeProtection: bool(keyVault.enablePurgeProtection)
+    softDeleteRetentionInDays: int(keyVault.softDeleteRetentionInDays)
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Allow'
+    }
+    privateEndpoints: [
+      {
+        service: 'vault'
```
[the remark this answers is not in the document]
Comment (kbabigumira) [reply infra/keyvault/application-keyvault.bicep:61]: Updated it, private endpoint name is now set

**Review by DaRoszko — APPROVED** [review]

**Review by aadev1 — APPROVED** [review]

**Pull request merged**